packages/cli/src/commands/init/init.ts

```diff
@@ -199,15 +193,28 @@ export default (async function initialize(
 ) {
   const root = process.cwd();
 
-  validateProjectName(projectName);
-
   /**
    * Commander is stripping `version` from options automatically.
    * We have to use `minimist` to take that directly from `process.argv`
    */
   const version: string = minimist(process.argv).version || DEFAULT_VERSION;
 
-  const directoryName = path.relative(root, options.directory || projectName);
+  // handles the case when '.' is passed as for projectDirectory
```

Comment:
AFAIK we don't need this special case.

1/ **We should keep `validateProjectName` where it was - on line 202**. All in all, the use case is always `react-native init MyProject --directory .` for those that want to initialise in the current directory. We should not allow `react-native init` to be run without a name (not sure if that's how it works in your PR) and then, overwrite the `projectNme`

2/ Do we actually need anything else? Does `directoryName` need to be `.`, not ``, hence special case treatment on line 207? I would imagine everything down the code works perfectly fine with an empty string, mind taking another pass at that?